shared_model/cryptography/ed25519_ursa_impl/crypto_provider.hpp

```diff
@@ -6,11 +6,16 @@
 #ifndef IROHA_URSA_CRYPTOPROVIDER_HPP
 #define IROHA_URSA_CRYPTOPROVIDER_HPP
 
+#if !defined(USE_LIBURSA)
+#error USE_LIBURSA must be defined
```

Comment:
This header can be unconditionally included, and cpp with dummy implementation can be used if the project is configured without Ursa.

Reply:
I don't like the dummy implementation approach:
- every place where this header included must be aware of that URSA might be unavailable and thus have the corresponding logic. this macro just makes compilation fail-fast, we can do just fine without it, but why not keep it if it helps?
- keeping a dummy implementation adds a risk to use it when it is not intended to, because it is indistinguishable from real implementation it substitutes
- dummy stub just looks a dirty approach to me